internal/ingress/controller/nginx.go

```diff
@@ -1084,37 +1074,65 @@ ratio = {{ .OtelSamplerRatio }}
 parent_based = {{ .OtelSamplerParentBased }}
 `
 
+func datadogOpentracingCfg(cfg ngx_config.Configuration) (string, error) {
+	jsn := map[string]interface{}{
+		"service": cfg.DatadogServiceName,
+		"agent_host": cfg.DatadogCollectorHost,
+		"agent_port": cfg.DatadogCollectorPort,
+		"environment": cfg.DatadogEnvironment,
+		"operation_name_override": cfg.DatadogOperationNameOverride,
+	}
+
+	// Omit "sample_rate" if the configuration's sample rate is unset (nil).
+	// Omitting "sample_rate" from the plugin JSON indicates to the tracer that
+	// it should use dynamic rates instead of a configured rate.
+	if cfg.DatadogSampleRate != nil {
+		jsn["sample_rate"] = *cfg.DatadogSampleRate
+	}
+
+	jsnBytes, err := json.Marshal(jsn)
+	if err != nil {
+		return "", err
+	}
+
+	return string(jsnBytes), nil
+}
+
+func opentracingCfgFromTemplate(cfg ngx_config.Configuration, tmplName string, tmplText string) (string, error) {
+	tmpl, err := template.New(tmplName).Parse(tmplText)
+	if err != nil {
+		return "", err
+	}
+
+	tmplBuf := bytes.NewBuffer(make([]byte, 0))
+	err = tmpl.Execute(tmplBuf, cfg)
+	if err != nil {
+		return "", err
+	}
+
+	return tmplBuf.String(), nil
+}
+
 func createOpentracingCfg(cfg ngx_config.Configuration) error {
-	var tmpl *template.Template
+	var fileContent string
 	var err error
 
 	if cfg.ZipkinCollectorHost != "" {
-		tmpl, err = template.New("zipkin").Parse(zipkinTmpl)
-		if err != nil {
-			return err
-		}
+		fileContent, err = opentracingCfgFromTemplate(cfg, "zipkin", zipkinTmpl)
 	} else if cfg.JaegerCollectorHost != "" || cfg.JaegerEndpoint != "" {
-		tmpl, err = template.New("jaeger").Parse(jaegerTmpl)
-		if err != nil {
-			return err
-		}
+		fileContent, err = opentracingCfgFromTemplate(cfg, "jaeger", jaegerTmpl)
 	} else if cfg.DatadogCollectorHost != "" {
-		tmpl, err = template.New("datadog").Parse(datadogTmpl)
-		if err != nil {
-			return err
-		}
+		fileContent, err = datadogOpentracingCfg(cfg)
 	} else {
-		tmpl, _ = template.New("empty").Parse("{}")
+		fileContent = "{}"
```

Comment:
Would it make sense to just return here and skip creating a file, or log some warning? That might be handled somewhere else

Reply:
Nah, I'd prefer to leave it as it was. The current behavior is to produce a file containing `{}`, you can even see it on disk in the image.